### PR TITLE
bug: fix error when instaling win64_llvm_mingw / Qt 6.7 and 6.8

### PR DESCRIPTION
**Pull request by miurahr**
ref #825

### CODE REVIEW

**Comment by miurahr** [comment]
Windows install-qt 6.7.3 win64_llvm_mingw for desktop failed as expected.

**Comment by miurahr** [comment]
```
2024-10-06 00:19:43,607 - aqt.main - ERROR - installer 1760 Updater caused an IO error: [Errno 2] No such file or directory: 'D:\\a\\1\\b\\Qt\\6.7.3\\llvm_mingw\\mkspecs\\qconfig.pri'
```

The target path became `llvm_mingw` but it was failed.

File: ci/generate_azure_pipelines_matrices.py [hunk 1]
```diff
@@ -136,6 +136,15 @@ def __init__(self, platform, build_jobs):
...
             "msvc2019_arm64",
+            is_autodesktop=True,  # Should install win64_msvc2019_arm64 in parallel
+        ),
+        BuildJob(
+            "install-qt",
+            "6.7.3",
+            "windows",
+            "desktop",
+            "win64_llvm_mingw",
+            "llvm_mingw",
```
Comment (miurahr) [inline ci/generate_azure_pipelines_matrices.py:147]:
```suggestion
            "llvm_mingw_64",
```

**Comment by miurahr** [comment]
failed
```
2024-10-06 01:06:05,578 - aqt.main - ERROR - installer 1208 Updater caused an IO error: [Errno 2] No such file or directory: 'D:\\a\\1\\b\\Qt\\6.7.3\\llvm_mingw_64\\mkspecs\\qconfig.pri'

```

File: aqt/metadata.py [hunk 2]
```diff
@@ -387,7 +387,7 @@ def dir_for_version(ver: Version) -> str:
 
     @staticmethod
     def get_arch_dir_name(host: str, arch: str, version: Version) -> str:
-        if arch.startswith("win64_mingw"):
+        if arch.startswith("win64") and arch.endswith("mingw"):
             return arch[6:] + "_64"
```
Comment (miurahr) [inline aqt/metadata.py:391]:
```suggestion
        if arch.startswith("win64_mingw"):
            return arch[6:] + "_64"
        elif arch.startswith("win64_llvm"):
            return "llvm-" + arch[11:] + "_64"    
```

File: ci/generate_azure_pipelines_matrices.py [hunk 3]
```diff
@@ -136,6 +136,15 @@ def __init__(self, platform, build_jobs):
...
             "msvc2019_arm64",
+            is_autodesktop=True,  # Should install win64_msvc2019_arm64 in parallel
+        ),
+        BuildJob(
+            "install-qt",
+            "6.7.3",
+            "windows",
+            "desktop",
+            "win64_llvm_mingw",
+            "llvm_mingw_64",
```
Comment (miurahr) [inline ci/generate_azure_pipelines_matrices.py:147]:
```suggestion
            "llvm-mingw_64",
```

**Comment by miurahr** [comment]
Patchings are worked as expected now.
```
2024-10-06 01:40:38,859 - aqt.updater - INFO - updater 4312 Patching D:\a\1\b\Qt\6.7.3\llvm-mingw_64\bin\qmake.exe
2024-10-06 01:40:38,859 - aqt.updater - INFO - updater 4312 Patching D:\a\1\b\Qt\6.7.3\llvm-mingw_64\lib\pkgconfig\Qt6Concurrent.pc
2024-10-06 01:40:38,859 - aqt.updater - INFO - updater 4312 Patching D:\a\1\b\Qt\6.7.3\llvm-mingw_64\lib\pkgconfig\Qt6Core.pc
2024-10-06 01:40:38,859 - aqt.updater - INFO - updater 4312 Patching D:\a\1\b\Qt\6.7.3\llvm-mingw_64\lib\pkgconfig\Qt6DBus.pc
2024-10-06 01:40:38,859 - aqt.updater - INFO - updater 4312 Patching D:\a\1\b\Qt\6.7.3\llvm-mingw_64\lib\pkgconfig\Qt6Designer.pc
2024-10-06 01:40:38,859 - aqt.updater - INFO - updater 4312 Patching D:\a\1\b\Qt\6.7.3\llvm-mingw_64\lib\pkgconfig\Qt6Gui.pc
2024-10-06 01:40:38,859 - aqt.updater - INFO - updater 4312 Patching D:\a\1\b\Qt\6.7.3\llvm-mingw_64\lib\pkgconfig\Qt6Help.pc
...
2024-10-06 01:40:38,905 - aqt.updater - INFO - updater 4312 Patching D:\a\1\b\Qt\6.7.3\llvm-mingw_64\lib\Qt6Widgets.prl
2024-10-06 01:40:38,905 - aqt.updater - INFO - updater 4312 Patching D:\a\1\b\Qt\6.7.3\llvm-mingw_64\lib\Qt6Xml.prl
```
but it is still  failed on build test.
```
Starting: build test with qmake w/o extra module
==============================================================================
Task         : PowerShell
Description  : Run a PowerShell script on Linux, macOS, or Windows
Version      : 2.245.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
==============================================================================
Generating script.
========================== Starting Command Output ===========================
"C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'D:\a\_temp\39650071-c659-4534-b38c-e81e4ac5672f.ps1'"
INFO    : aqtinstall(aqt) v3.1.19.dev24 on Python 3.12.6 [CPython MSC v.1940 64 bit (AMD64)]
ERROR   : The package 'qt.tools.' was not found while parsing XML of package information!
==============================Suggested follow-up:==============================
* Please use 'aqt list-tool windows desktop tools_mingw' to show tool variants available.
Failed to install tools_mingw
At D:\a\_temp\39650071-c659-4534-b38c-e81e4ac5672f.ps1:25 char:5
+     throw 'Failed to install tools_mingw'
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Failed to install tools_mingw:String) [], RuntimeException
    + FullyQualifiedErrorId : Failed to install tools_mingw
```

@pcolby could you help to investigate it?

**Comment by miurahr** [comment]
```
 python -m aqt list-tool windows desktop tools_mingw
qt.tools.mingw47
qt.tools.win32_mingw48
qt.tools.win32_mingw482
qt.tools.win32_mingw491
qt.tools.win32_mingw492
qt.tools.win32_mingw530
qt.tools.win32_mingw730
qt.tools.win32_mingw810
qt.tools.win64_mingw730
qt.tools.win64_mingw810
```
No entry for 
- https://download.qt.io/online/qtsdkrepository/windows_x86/desktop/tools_llvm_mingw1706/
- https://download.qt.io/online/qtsdkrepository/windows_x86/desktop/tools_mingw1310/
@ddalcino  any suggestions?

**Comment by miurahr** [comment]
@pcolby Could you look at? https://github.com/miurahr/aqtinstall/blob/master/ci/steps.yml#L274

**Comment by ddalcino** [comment]
> No entry for
> * https://download.qt.io/online/qtsdkrepository/windows_x86/desktop/tools_llvm_mingw1706/
> * https://download.qt.io/online/qtsdkrepository/windows_x86/desktop/tools_mingw1310/
> @ddalcino any suggestions?


Looks all right to me.

```shell
$ python -m aqt list-tool windows desktop | grep mingw
tools_mingw1310
tools_mingw90
tools_mingw
tools_llvm_mingw1706

$ python -m aqt list-tool windows desktop tools_llvm_mingw1706
qt.tools.win64_llvm_mingw1706

$ python -m aqt list-tool windows desktop tools_mingw1310  
qt.tools.win64_mingw1310
```

**Comment by ddalcino** [comment]
#829 should make this work

**Comment by miurahr** [comment]
I've pulled @ddalcino changes to the topic branch. Thanks.

File: ci/generate_azure_pipelines_matrices.py [hunk 4]
```diff
@@ -136,6 +158,16 @@ def __init__(self, platform, build_jobs):
...
+        ),
+        BuildJob(
+            "install-qt",
+            "6.7.3",
+            "windows",
+            "desktop",
+            "win64_llvm_mingw",
+            "llvm-mingw_64",
+            mingw_variant="win64_llvm_mingw1706",
             is_autodesktop=True,    # Should install win64_msvc2019_arm64 in parallel
```
Comment (hasselmm) [inline ci/generate_azure_pipelines_matrices.py:171]: Is this correct? Does the llvm toolchain really need the msvc toolchain in parallel?

Comment (ddalcino) [reply ci/generate_azure_pipelines_matrices.py:171]: Good catch. The autodesktop flag is for automatically installing the desktop architecture of Qt in parallel with an android or wasm architecture of Qt. I think the flag will be ignored in this context. Likely a copy-paste error, but benign.

Comment (miurahr) [inline ci/generate_azure_pipelines_matrices.py:171]:
```suggestion
            is_autodesktop=False,
```
Ignore autodesktop for here.

**Pull request merged**